web/frontend/src/components/HeartbeatIndicator.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+
+/**
+ * Component to display all received heartbeats, marking the active node and if they are considered alive (received heartbeat within the last 3s).
+ * @param {Record<string, { sender_id: string; tick: number; active: boolean; timestamp: number }>} receivedHeartbeats - Record containing the last heartbeats received in the network.
+ */
+export default function HeartbeatIndicator({
+  receivedHeartbeats,
+}: {
+  receivedHeartbeats: Record<string, { sender_id: string; tick: number; active: boolean; timestamp: number }>;
+}) {
+  // Tracking now to rerender every second
+  const [now, setNow] = useState(Date.now() / 1000);
+  // Defining the maximum time in seconds to pass between two subsequent heartbeats before the node is consideres unalive
+  const maxTimeBetweenHeartbeats = 3;
+
+  useEffect(() => {
+    const interval = setInterval(() => setNow(Date.now() / 1000), 1000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
+  const nodeHeartbeats = Object.values(receivedHeartbeats).map((heartbeat, index) => {
+    const nodeOnline = now - heartbeat.timestamp < maxTimeBetweenHeartbeats;
```

Comment:
I would not rely on time sync between the time stamp of the node on the pi and any other device.
Maybe wait for the first message and use its timestamp to calculate the offset between the clocks. This will of cause not account for clock drift.
Maybe I am reading this code wrong but I will not learn JS for this.

Reply:
Now, you're actually right. This is the bug I looked for. I will fix this by saving my own system time on receive of the message. 
This is TS and not JS btw. ;)